src/components/Filters/BooleanFilter.tsx

```diff
@@ -0,0 +1,60 @@
+import { Filter } from "src/components/Filters/types";
+import { SelectField } from "src/inputs/SelectField";
+import { defaultLabel } from "src/utils/defaultLabel";
+import { defaultTestId } from "src/utils/defaultTestId";
+import { TestIds } from "src/utils/useTestIds";
+
+export type BooleanOption = [boolean | undefined, string];
+
+export interface BooleanFilterProps {
+  options?: BooleanOption[];
+  label?: string;
+  defaultValue?: undefined | boolean;
+}
+
+const defaultBooleanOptions: BooleanOption[] = [
+  [undefined, "Any"],
+  [true, "Yes"],
+  [false, "No"],
+];
+
+export function booleanFilter(opts: BooleanFilterProps): (key: string) => Filter<boolean> {
```

Comment:
Each filter type is in its own file, and has both the rendering + value updating logic for it, instead of those being spread around in misc places.


Also the `fooFilter` functions technically return a lambda, i..e the `key: string => ...` which is so that the filter classes can take `key` as a constructor, i.e.: `{ marketId: singleSelect(...) }` will be passed `marketId` so that the `SingleFilter` class can use `marketId` primarily to create the default label.